modules/packages/Socket.chpl

```diff
@@ -461,14 +718,14 @@ proc tcpListener.accept(in timeout: timeval = new timeval(-1,0)):tcpConn throws
         throw SystemError.fromSyserr(err_out, "accept() failed");
       }
       // no indefinitely blocking wait
-      if timeout.tv_sec != -1 {
-        var totalTimeout = timeout.tv_sec*1000000 + timeout.tv_usec;
+      if timeout.tv_sec:int != -1 {
+        var totalTimeout = timeout.tv_sec:c_long*1000000 + timeout.tv_usec:c_long;
```

Comment:
It seems inconsistent that we cast to `int` in some cases as `c_long` in others when converting these timeval fields to Chapel code.  Did you have a rationale for preferring one or the other in various cases?

Reply:
I think in the back of my mind, I was assuming that casting the `time_t` to `int` for comparisons would avoid an additional implicit cast. (`t:int == -1` felt more direct than `t:c_long == -1:c_long`)

And I used `c_long` in the other cases, because from what I read, `time_t` and `suseconds_t` are *usually* implemented as a `long` in c, so that also seemed more direct. 

But consistency is probably better, so I'll make those all `c_long`

Reply:
> felt more direct than t:c_long == -1:c_long)

Note that `-1:c_long` isn't necessary because `-1` is a param value in Chapel, and param integer values can coerce to all other integer types that the value can fit within.  So as long as `-1` is a legal c_long value (and of course it is :) ), it will automatically coerce to that (at compile-time) without an explicit cast.

Reply:
Ok got it! I knew the explicit cast wasn't necessary, but I had assumed the *implicit* cast would happen at runtime.